paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommitImpl.java

```diff
@@ -694,10 +707,14 @@ public boolean tryCommitOnce(
             Snapshot.CommitKind commitKind,
             @Nullable Snapshot latestSnapshot,
             @Nullable Long safeLatestSnapshotId,
+            String branchName,
             @Nullable String newStatsFileName) {
         long newSnapshotId =
                 latestSnapshot == null ? Snapshot.FIRST_SNAPSHOT_ID : latestSnapshot.id() + 1;
-        Path newSnapshotPath = snapshotManager.snapshotPath(newSnapshotId);
+        Path newSnapshotPath =
+                branchName.equals(MAIN_BRANCH)
```

Comment:
We can compare branch name and get path for table in BranchManager.getBranchPath instead of here

Reply:
It will call the getBranchPath method.
![image](https://github.com/apache/incubator-paimon/assets/45089228/1f215d3d-7e04-41c2-a2b8-2d4c222ca7f7)